amethyst_renderer/src/pipe/stage.rs

```diff
@@ -67,73 +47,111 @@ impl Stage {
     pub fn is_enabled(&self) -> bool {
         self.enabled
     }
+}
+
+pub trait StageLength {
+    /// Get number of encoders needed for this stage.
+    fn encoders_required(jobs_count: usize) -> usize;
+}
+
+impl<L> StageLength for Stage<L> 
+    where L: Length
+{
+    /// Get number of encoders needed for this stage.
+    #[inline]
+    fn encoders_required(jobs_count: usize) -> usize {
+        use std::cmp;
+        cmp::max(L::len() * jobs_count, 1)
+    }
+}
+
+
+pub trait StageApply<'a, D> {
+    type Apply: IntoParallelIterator<Item=()>;
+    fn apply(&'a self, encoders: &'a mut [Encoder], jobs_count: usize, data: D) -> Self::Apply;
+}
+
+
+
```

Comment:
Could you please reduce the number of subsequent empty lines to 1? I can see why some people use two of them, but 4?

Reply:
If this is biggest issue in this PR I call it success 😄